### DISCUSSION TITLE
confusion when absolute/relative position nested and combined

### DISCUSSION THREAD

**Discussion by tian-yan** [tailwindlabs/tailwindcss#4565] [Help] [answered]
I came across some conflicting and confusing scenarios when playing around nested absolute/relative positions. 
My understanding for absolute position is as below:
>- **`position: absolute`** without any **`top | bottom | left | right`** statement,  the element will be placed in top-left corner of its nearest positioned **[`position: (none static)`]** Parent (if not exist, then Grandfather and all the way up, but not Uncles) element.
>- **`position: absolute`** with **`top | bottom | left | right`** statement, the element will be offset relative to its nearest positioned **[`position: (none static)`]** Parent (if not exist, then Grandfather and all the way up, but not Uncles) element.

However, above conflicts with my testing in TailwindCSS Play, 
1. No `top | bottom | left | right` at all anywhere in [demo code #1](https://play.tailwindcss.com/YsJs8tNEjm?size=676x720), Why **"Child (absolute)"** is not placed in top-left corner of **"Father Container(absolute)"** when **"Father Container (absolute)"** is in the top-left corner of **"Grandfather Container (relative)"** as expected ?

2. Still No `top | bottom | left | right` in [demo code #2](https://play.tailwindcss.com/23bDd2E0bN?size=676x720), when I change **"Father Container"** to **`position: static (default value)`**,  why **"Child (absolute)"** is not in the top-left corner of **"Grandfather Container (relative)"** as I expected ?

**Comment by VicGUTT** [comment, marked as answer]
Hey ! Yes positioning can be confusing/interesting.
Your above assumptions are correct but incomplete in my opinion as it leaves out sibling (brother/sister) elements.

### Simplified scenario

What happens when an absolutely positioned element, that does not have any placement guidance (top, right...), is within a relatively positioned parent, and have siblings ? Where should that element be positioned within the parent and how should the surrounding content reflow ?

Here's the above scenario translated to code for better visualisation:

```html
<main class="relative">
    <p>Older brother</p>
    <p class="absolute">Middle kid</p>
    <p>The cute little one</p>
</main>
```

From this [Tailwind Play demo](https://play.tailwindcss.com/KQsmgZVX1O) you can see how the "_**Older brother**_" is unbothered and unaffected by the  "_**Middle kid**_" being absolutely positioned. "_**The cute little one**_" on the other hand has to reflow to fill the space leaved by the  "_**Middle kid**_".
If however you add placement directives to the absolutely positioned child (`<p class="absolute top-0">Middle kid</p>`) then those explicit directives will be honored.

In other words, from what I can tell, absolutely positioned elements does not seem to affect any naturally flowed elements that comes before it. The elements after it however has to reflow, unless of course they  are also taken out of the natural flow (positioned absolutely, fixed...).


### Your demoes

Both of your demoes, if you think about it, are the same and fall into the same simplified scenario as above.
If in demo 1 for example you switched things up to be like this:

```html
<nav class="absolute bg-yellow-700 text-center">
    <!-- Child Container with absolute position -->
    <ul class="absolute bg-gray-200">
        <li><a href="#">Child (absolute)</a></li>
    </ul>

    <input type="checkbox" id="Side_Menu" class="hidden">
    <label for="Side_Menu" class="text-white text-l font-bold">
        <span>Father Container (absolute)</span>
    </label>
</nav>
```

or like this:

```html
<nav class="absolute bg-yellow-700 text-center">
    <input type="checkbox" id="Side_Menu" class="hidden">
    <label for="Side_Menu" class="text-white text-l font-bold">
        <span>Father Container (absolute)</span>
    </label>

    <!-- Child Container with absolute position -->
    <ul class="absolute top-0 bg-gray-200">
        <li><a href="#">Child (absolute)</a></li>
    </ul>
</nav>
```

then you would have the expected behavior you outlined for the "_**Child (absolute)**_".
Your assumption of "_**Father Container (absolute)**_" being in the top-left corner of "_**Grandfather Container (relative)**_" though has been corrupted by the use of _flexbox_ on the grand-parent. If you remove the `flex` class the parent will be beneath the grand-parent and again follow the same scenario outline in my "Simplified scenario" above.


### Side note

Some HTML and Tailwind errors I've noticed:

- The `<banner>` element does not exist in HTML. Maybe you meant `<header>` ?
- There's an extra `</li>` after `<li><a href="#">Child (absolute)</a></li>`.
- The class `text-l` in `<label for="Side_Menu" class=" text-white text-l font-bold">` does not exist in Tailwind, you probably meant `text-lg` or `text-xl` ?

---

Hope I helped clear things out :v:

**Reply by tian-yan** [reply]
Thanks for your time on the detailed answer.  I took some time to digest your insights and verified them via tailwind css play as well. I believe we are safe to conclude that, for `position: absolute` elements,
- if no placement guidance, it will stay where it was originally but shrink its size to its contents unless explicitly specified width/height, and its following elements will re-flow.

- when any `top | bottom | left | right` specified, it will move relative to its nearest `positioned` ancestor.
